### PR TITLE
iproute2: bump to 4.11

### PR DESCRIPTION
**Pull request by kdarbyshirebryant**
Signed-off-by: Kevin Darbyshire-Bryant <kevin@darbyshire-bryant.me.uk>

### CODE REVIEW

**Comment by LynxChaus** [comment]
Did you test it?
source file hash not match: 22b1e1c1fc704ad35837e5a66103739727b8b48ac90b48c13f79b7367ff0a9a8 from 4.10.0, correct 72671028bda696d0cb8f48ec8e702581c3a501caeed33eec3a81d7041cbc8026,
patch 950-add-cake-to-tc.patch damaged, first hunk not apply (mismatched changed lines 61 <> 60).

File: package/network/utils/iproute2/Makefile [hunk 1]
```diff
@@ -49,7 +51,7 @@ endef
 define Package/tc
 $(call Package/iproute2/Default)
   TITLE:=Traffic control utility
-  DEPENDS:=+kmod-sched-core
+  DEPENDS:=+kmod-sched-core +libxtables +libmnl
```
Comment (hauke) [inline package/network/utils/iproute2/Makefile]: libmnl and libxtbales are optional dependencies, which features do you need?

File: package/network/utils/iproute2/Makefile [hunk 2]
```diff
@@ -72,6 +74,12 @@ $(call Package/iproute2/Default)
   TITLE:=Network statistics utility
 endef
 
+define Package/devlink
+$(call Package/iproute2/Default)
+  TITLE:=Network devlink utility
+  DEPENDS:=+libmnl
+endef
```
Comment (hauke) [inline package/network/utils/iproute2/Makefile]: yes for devlink you bee libmnl

File: package/network/utils/iproute2/patches/001-config.patch [hunk 3]
```diff
@@ -1,7 +1,8 @@
 --- /dev/null
 +++ b/Config
-@@ -0,0 +1,4 @@
+@@ -0,0 +1,5 @@
 +# Fixed config to disable ATM support even if present on host system
 +TC_CONFIG_ATM:=n
 +TC_CONFIG_ACTION_GACT=y
 +TC_CONFIG_ACTION_PROB=y
```
Comment (hauke) [inline package/network/utils/iproute2/patches/001-config.patch]: TC_CONFIG_ACTION_GACT=y and TC_CONFIG_ACTION_PROB=y do not exist any more.
Please add TC_CONFIG_XT:=y which would be added in case xtables is available

File: package/network/utils/iproute2/Makefile [hunk 4]
```diff
@@ -90,11 +98,12 @@ TARGET_CFLAGS += -DHAVE_SETNS
 TARGET_CFLAGS += -ffunction-sections -fdata-sections
```
Comment (hauke) [inline package/network/utils/iproute2/Makefile]: please add IP_CONFIG_SETNS to Concifg file instead of manually activating HAVE_SETNS here. Normally the Config file gets generated with such information.

File: package/network/utils/iproute2/patches/999-honor-copts.patch [hunk 5]
```diff
@@ -0,0 +1,11 @@
+--- a/Makefile
++++ b/Makefile
+@@ -49,7 +49,7 @@ CCOPTS = -O2 $(EXTRA_CCOPTS)
+ WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
+ WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
+ 
+-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS)
++CFLAGS := $(WFLAGS) -I../include $(EXTRA_CCOPTS) $(DEFINES) $(CFLAGS)
```
Comment (hauke) [inline package/network/utils/iproute2/patches/999-honor-copts.patch]: please merge this together with 110-extra-ccopts.patch it does a very similar think in a different line.

File: package/network/utils/iproute2/Makefile [hunk 6]
```diff
@@ -6,15 +6,17 @@
 #
 
 include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
```
Comment (hauke) [inline package/network/utils/iproute2/Makefile]: why is this needed?

**Comment by kdarbyshirebryant** [comment]
Guys, thanks for the review.  The reality is that my $CLUE=NULL when it comes to iproute2.  Russell Senior had a bump to 4.10 that he was working on a month or two ago which since went quiet.  I wanted to make sure this work wasn't lost as it's a bit of a challenge getting it to compile with musl and see if I could get it to current level 4.11.  I had a number of little tweaks related to 'cake' in my own tree, hence the flustercluck with non applying patches and somehow the package HASH when I cherry-picked it as a single commit for this PR.  I'll update the PR addressing the points I understand once I can get the thing to compile again :-)

File: package/network/utils/iproute2/patches/999-honor-copts.patch (hunk 5 above)
Comment (kdarbyshirebryant) [inline package/network/utils/iproute2/patches/999-honor-copts.patch]: Actually I think the entire 999 patch is bogus since 110-extra-ccopts does
CCOPTS = -O2 $(EXTRA_CCOPTS)

So CCOPTS already has $EXTRA_CCOPTS.

File: package/network/utils/iproute2/Makefile (hunk 6 above)
Comment (kdarbyshirebryant) [inline package/network/utils/iproute2/Makefile]: No idea - I'll try removing it and see what happens.

File: package/network/utils/iproute2/Makefile (hunk 4 above)
Comment (kdarbyshirebryant) [inline package/network/utils/iproute2/Makefile]: Done

**Comment by nbd168** [comment]
Merged to my staging tree, thanks
